pkg/internal/server/grpc/common_test.go

```diff
@@ -141,96 +203,50 @@ func (th *testHarness) GRPCHealthClient(t *testing.T) healthpb.HealthClient {
-	cbtClient := fakecbt.NewSimpleClientset()
-	cbtClient.PrependReactor("get", "snapshotmetadataservices", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
-		ga := action.(clientgotesting.GetAction)
-		if ga.GetName() != th.DriverName {
-			return false, nil, nil
-		}
-		sms := &smsv1alpha1.SnapshotMetadataService{
-			ObjectMeta: apimetav1.ObjectMeta{
-				Name: th.DriverName,
-			},
-			Spec: smsv1alpha1.SnapshotMetadataServiceSpec{
-				Audience: th.Audience,
-			},
-		}
-		return true, sms, nil
-	})
-
-	return cbtClient
+	return th.fakeCBTClient
 }
 
 func (th *testHarness) FakeSnapshotClient() *fakesnapshot.Clientset {
-	snapshotClient := fakesnapshot.NewSimpleClientset()
-	snapshotClient.PrependReactor("get", "volumesnapshots", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
-		ga := action.(clientgotesting.GetAction)
-		vs := &snapshotv1.VolumeSnapshot{
-			ObjectMeta: apimetav1.ObjectMeta{
-				Name:      ga.GetName(),
-				Namespace: ga.GetNamespace(),
-			},
-			Spec: snapshotv1.VolumeSnapshotSpec{
-				VolumeSnapshotClassName: stringPtr("csi-snapshot-class"),
-				Source: snapshotv1.VolumeSnapshotSource{
-					PersistentVolumeClaimName: stringPtr("pvc-1"),
-				},
-			},
-			Status: &snapshotv1.VolumeSnapshotStatus{
-				ReadyToUse:                     boolPtr(true),
-				BoundVolumeSnapshotContentName: stringPtr("vs-content-1"),
-			},
-		}
-		return true, vs, nil
-	})
-	snapshotClient.PrependReactor("get", "volumesnapshotcontents", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
-		ga := action.(clientgotesting.GetAction)
-		vs := &snapshotv1.VolumeSnapshotContent{
-			ObjectMeta: apimetav1.ObjectMeta{
-				Name:      ga.GetName(),
-				Namespace: ga.GetNamespace(),
-			},
-			Spec: snapshotv1.VolumeSnapshotContentSpec{
-				Driver: th.DriverName,
-			},
-			Status: &snapshotv1.VolumeSnapshotContentStatus{
-				ReadyToUse:     boolPtr(true),
-				SnapshotHandle: stringPtr("snap-id-1"),
+	return th.fakeSnapshotClient
+}
+
+func volumeSnapshot(name, namespace string) *snapshotv1.VolumeSnapshot {
+	return &snapshotv1.VolumeSnapshot{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: snapshotv1.VolumeSnapshotSpec{
+			VolumeSnapshotClassName: stringPtr("csi-snapshot-class"),
+			Source: snapshotv1.VolumeSnapshotSource{
+				PersistentVolumeClaimName: stringPtr("pvc-1"),
 			},
-		}
-		return true, vs, nil
-	})
+		},
+		Status: &snapshotv1.VolumeSnapshotStatus{
+			ReadyToUse:                     boolPtr(true),
+			BoundVolumeSnapshotContentName: stringPtr(name),
+		},
+	}
+}
 
-	return snapshotClient
+func volumeSnapshotContent(name, namespace, driver string) *snapshotv1.VolumeSnapshotContent {
+	return &snapshotv1.VolumeSnapshotContent{
+		ObjectMeta: apimetav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
```

Comment:
VolumeSnapshotContent objects do not use `namespace` in ObjectMeta.

The `Spec.VolumeSnapshotRef` field uses `name` and `namespace` to identify the VolumeSnapshot object.